packages/gateway/gateway.config.ts

```diff
@@ -0,0 +1,36 @@
+import { defineConfig } from '@graphql-hive/gateway';
+
+export const gatewayConfig = defineConfig({
+  supergraph: '../../supergraph.graphql',
+  // Configure header propagation to forward auth headers to subgraphs
+  propagateHeaders: {
+    fromClientToSubgraphs({ request }) {
+      const authHeader = request.headers.get('authorization');
+      // Only return the header if it exists
+      if (authHeader) {
+        return {
+          authorization: authHeader,
+        };
+      }
+      // Return void if no auth header
+      return;
+    },
+  },
+  // Configure generic auth for the Gateway
+  genericAuth: {
```

Comment:
`genericAuth` is super broad. Why do we delegate the authentication process as-is to the subgarph? 
It should be done once, in the gateway, and then the subgraphs only get what they really needed.